shell/client/_widgets.scss

```diff
@@ -1,3 +1,194 @@
+}
+
+.button-secondary3 {
+  @extend %button-large;
+  @extend %button-secondary;
+}
+
+.button-secondary4 {
+  @extend %button-xlarge;
+  @extend %button-secondary;
+}
+
+.button-tertiary1 {
+  @extend %button-small;
+  @extend %button-tertiary;
+}
+
+.button-tertiary2 {
+  @extend %button-medium;
+  @extend %button-tertiary;
+}
+
+.button-tertiary3 {
+  @extend %button-large;
+  @extend %button-tertiary;
+}
+
+.button-tertiary4 {
+  @extend %button-xlarge;
+  @extend %button-tertiary;
+}
+
+
+.search-row1 {
+  position: relative;
+  @media #{$desktop} {
+    margin-bottom: 24px;
+    width: 100%;
+    vertical-align: top;
+  }
+  @media #{$mobile} {
+    margin-bottom: 0;
+    display: inline-block;
+    width: calc(100% - 100px);
+    vertical-align: middle;
+  }
+  >label {
+    display: inline-block;
+    >span.search-icon {
+      @extend %pseudo-img-tag;
+      background-image: url("/search.svg");
+      background-size: 24px 24px;
+      background-position: center;
+      display: inline-block;
+      box-sizing: border-box;
+      width: 32px;
+      height: 32px;
+      vertical-align: top;
+    }
+    >input.search-bar {
+      display: inline-block;
+      vertical-align: top;
+      width: 33%;
+      padding: 0px 5px;
+      font-size: 14pt;
+      height: 32px;
+      background-color: $grainlist-searchbar-background-color;
+      border: 1px solid $grainlist-searchbar-outline-color;
+      box-sizing: border-box;
+      &:focus {
+        border: 1px solid $grainlist-searchbar-outline-color-focus;
+      }
+      @media #{$desktop} {
+        width: calc(100% - 36px);
+      }
+      @media #{$mobile} {
+        width: calc(100% - 60px);
+      }
+    }
+    @media #{$desktop} {
+      width: 50%;
+    }
+  }
+}
+
+.details1 {
+  color: #444444;
+  font-size: 85%;
+}
+
+%button-small {
+  @extend %button-margin;
+  border-radius: 4px;
+  padding: 4px 8px;
+}
+
+%button-medium {
+  @extend %button-margin;
+  font-size: 110%;
+  border-radius: 4px;
+  padding: 6px 10px;
+}
+
+%button-large {
+  @extend %button-margin;
+  font-size: 130%;
+  border-radius: 4px;
+  padding: 8px 12px;
+}
+
+%button-xlarge {
+  @extend %button-margin;
+  font-size: 150%;
+  font-weight: 600;
+  border-radius: 4px;
+  padding: 10px 16px;
+}
+
+%button-primary {
+  color: #ffffff;
+  background-color: #65468e;
+  border: 1px solid darken(#57427f, 10%);
+  &:hover {
+    background-color: #714daa;
+  }
+  &:disabled {
+    @extend %button-disabled;
+  }
+}
+
+%button-secondary {
+  color: #000000;
+  background-color: #ffffff;
+  border: 1px solid darken(#d1d1d1, 10%);
+  &:hover {
+    background-color: #f8f8f8;
+  }
+  &:disabled {
+    @extend %button-disabled;
+  }
+}
+
+%button-tertiary {
+  color: #000000;
+  background-color: #e2e2e2;
+  border: 1px solid darken(#afafaf, 10%);
+  &:hover {
+    background-color: #eaeaea;
+  }
+  &:disabled {
+    @extend %button-disabled;
+  }
+}
+
+%button-margin {
+  margin: 10px 5px;
+}
+
+%button-disabled {
```

Comment:
Two comments on this:

* Disabled buttons should probably be lower contrast, to make it clear that they are disabled, especially when you're looking at a form with a single disabled button in isolation. `#777777` for the foreground color and `lighten(#afafaf, 10%)` for the border looked decent to me; other ideas totally welcome.
* Disabled buttons should also override `&:hover` (with the same values) so that disabled buttons don't show their "other" colors on mouseover.